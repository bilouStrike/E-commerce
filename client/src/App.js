import React, { useEffect } from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import './App.css';
import HomePage from './pages/homepage/homepage.component';
import ShopPage from './pages/shop/shop.component';
import CheckoutPage from './pages/checkout/checkout.component';
import SignInSignUp from './pages/signIn-signup-page/signIn-signUp.component';
import Header from './components/header/header.component';
import { selectCurrentUser } from './redux/user/user.selectors';
import { checkUserSession } from './redux/user/user.actions';

export const App = ({ checkUserSession, currentUser }) => {

    useEffect(()=> {
      checkUserSession()
    }, [checkUserSession]);

    return (
      <div className="app-container">
        <Header/>
        <Switch>
          <Route exact path='/' component={HomePage} />
          <Route path='/shop' component={ShopPage} />
          <Route exact path='/checkout' component={CheckoutPage} />
          <Route exact path='/signin' render={ () => currentUser ? ( <Redirect to='/'/> ) : ( <SignInSignUp/>  )} />
        </Switch>
      </div>
    );
};

const mapStateToProps = ( state ) => ({
  currentUser: selectCurrentUser(state)
});

const mapDispatchToProps = dispatch => ({
  checkUserSession: () => dispatch(checkUserSession())
});

export default connect(mapStateToProps, mapDispatchToProps)(App);
