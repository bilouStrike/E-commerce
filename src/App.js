import React from 'react';
import { Switch, Route, Redirect } from 'react-router-dom';
import { connect } from 'react-redux';

import './App.css';

import HomePage from './pages/homepage/homepage.component';
import ShopPage from './pages/shop/shop.component';
import CheckoutPage from './pages/checkout/checkout.component';

import SignInSignUp from './pages/signIn-signup-page/signIn-signUp.component';
import Header from './components/header/header.component';
import { selectCurrentUser } from './redux/user/user.selectors';

class App extends React.Component {

  unsubscribeFromAuth = null;

  componentDidMount() {
    /*this.unsubscribeFromAuth = auth.onAuthStateChanged( async userAuth => {
        if( userAuth ) {
          const userRef = await createUserProfileDocument(userAuth);
          userRef.onSnapshot( snapShot => {
            setCurrentUser({
                  currentUser: {
                      id: snapShot.id,
                      ...snapShot.data()
                  }
              })
          });

        } else {
          setCurrentUser({currentUser:userAuth});
        }
      }
    )*/
  }

  componentWillUnmount() {
    this.unsubscribeFromAuth();
  }

  render() {
    console.log(this.props.currentUser);
    return (
      <div className="app-container">
        <Header/>
        <Switch>
          <Route exact path='/' component={HomePage} />
          <Route path='/shop' component={ShopPage} />
          <Route exact path='/checkout' component={CheckoutPage} />
          <Route exact path='/signin' render={ () => this.props.currentUser ? ( <Redirect to='/'/> ) : ( <SignInSignUp/>  )} />
        </Switch>
      </div>
    );
  }
}

const mapStateToProps = ( state ) => ({
  currentUser: selectCurrentUser(state)
})

export default connect(mapStateToProps)(App);
