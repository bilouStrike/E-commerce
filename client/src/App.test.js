import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Redirect } from 'react-router-dom';
import React from 'react';
import HomePage from './pages/homepage/homepage.component';
import ShopPage from './pages/shop/shop.component';
import CheckoutPage from './pages/checkout/checkout.component';
import SignInAndSignUpPage from './pages/signIn-signup-page/signIn-signUp.component';

import { App } from './App.js';

describe('App', () =>{
    let wrapper;
    beforeEach( () =>{
        wrapper = shallow(<App />);
    });

    it('Should render App component', () => {
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('Should route / to homePage', () => {
        expect(wrapper.find('Route[exact=true][path="/"]').first().prop('component')).toBe(HomePage);
    });

    it('Should route /shop to ShopPage', () => {
        expect(wrapper.find('Route[path="/shop"]').first().prop('component')).toBe(ShopPage);
    });

    it('Should route /checkout to CheckOut', () => {
        expect(wrapper.find('Route[path="/checkout"]').first().prop('component')).toBe(CheckoutPage);
    });

    it('Should route /signin to SignInSignUp if currentUser false', () => {
        const mockCurrentUser = false;
        const wrapperwithUserTrue = shallow(<App currentUser={mockCurrentUser} />);
        expect(wrapperwithUserTrue.find('Route[path="/signin"]').at(0).props().render()).toEqual(<SignInAndSignUpPage/>);
    });

    it('Should route /signin to /Homepage if currentUser true', () => {
        const mockCurrentUser = true;
        const wrapperwithUserFalse = shallow(<App currentUser={mockCurrentUser} />);
        expect(wrapperwithUserFalse.find('Route[path="/signin"]').at(0).props().render()).toEqual(<Redirect to='/'/>);
    });
 });
