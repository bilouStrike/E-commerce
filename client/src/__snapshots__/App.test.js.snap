// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Should render App component 1`] = `
<div
  className="app-container"
>
  <Connect(Header) />
  <Switch>
    <Route
      component={[Function]}
      exact={true}
      path="/"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(ShopPage)",
          "type": [Function],
        }
      }
      path="/shop"
    />
    <Route
      component={
        Object {
          "$$typeof": Symbol(react.memo),
          "WrappedComponent": [Function],
          "compare": null,
          "displayName": "Connect(CheckoutPage)",
          "type": [Function],
        }
      }
      exact={true}
      path="/checkout"
    />
    <Route
      exact={true}
      path="/signin"
      render={[Function]}
    />
  </Switch>
</div>
`;
